block/manager.go

```diff
@@ -180,28 +180,26 @@ func getInitialState(ctx context.Context, genesis genesis.Genesis, signer signer
 			}}
 
 		var signature types.Signature
-		var pubKey crypto.PubKey
 		// The signer is only provided in aggregator nodes. This enables the creation of a signed genesis header,
 		// which includes a public key and a cryptographic signature for the header.
 		// In a full node (non-aggregator), the signer will be nil, and only an unsigned genesis header will be initialized locally.
 		if signer != nil {
-			pubKey, err = signer.GetPublic()
+			signature, err = types.GetSignature(header, signer)
 			if err != nil {
-				return types.State{}, fmt.Errorf("failed to get public key: %w", err)
+				return types.State{}, fmt.Errorf("failed to get header signature: %w", err)
 			}
-			signature, err = getSignature(header, signer)
+		} else {
+			// If the signer is not provided, create a noop signer from the proposer address
+			signer, err = noopsigner.NewNoopSignerFromAddress(genesis.ProposerAddress)
 			if err != nil {
-				return types.State{}, fmt.Errorf("failed to get header signature: %w", err)
+				return types.State{}, fmt.Errorf("failed to create noop signer: %w", err)
```

Comment:
we should try to avoid this. Proposer address should always be present. Otherwise fail

Reply:
this seems very hacky. signer is meant for signing. Noopsigner is for testing, not for production. why cant we use the signer how it was before? It seems like that part wasnt the issue